pytorch_lightning/trainer/trainer.py

```diff
@@ -1431,58 +1437,109 @@ def _call_teardown_hook(self) -> None:
         # summarize profile results
         self.profiler.describe()
 
-    def call_hook(
```

Comment:
will you deprecate this one?

Reply:
because the new `_call_hook` functions are replacing it completely. There are no more calls of `call_hook` left in the codebase

Reply:
I am just removing it, not deprecating, since is it meant to be used only in our internal codebase

Reply:
@daniellepintz to be safe, I think we should go with the deprecation out of precaution. This deprecation could be handled in a separate PR to keep this one focused on introducing the granular hook funtions & improved profiling